### PR TITLE
Templates page for site

### PR DESCRIPTION
**Pull request by anirudh-tantry**

### CODE REVIEW

File: layouts/default.vue [hunk 1]
```diff
@@ -12,6 +12,13 @@ import Footer from '../components/Footer.vue'
 
 export default {
   components: { Footer, Navbar },
+  computed: {
+    isTemplateDetailPage() {
+      return (
+        this.$route.path.startsWith('/templates/') && this.$route.params?.slug
+      )
+    },
+  },
```
Comment (ankitsinghaniyaz) [inline layouts/default.vue:21]: let's create a new layout, that's a better architecture and let's use that on templates page

File: pages/templates/_slug.vue [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        const { data } = await this.$axios.get(
+          `https://app.formester.com/templates/${this.$route.params.slug}.json`
+        )
+        this.template = data
+      } catch (error) {
+        this.template = null
+      }
+    },
+    redirectTo(id) {
+      window.location.href = `https://app.formester.com/forms/new?template_id=${id}`
```
Comment (ankitsinghaniyaz) [inline pages/templates/_slug.vue:87]: let's open this in new tab

**Review by ankitsinghaniyaz — APPROVED** [review]

File: pages/templates/index.vue [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+export default {
+  components: { Templates },
+  computed: {
+    meta() {
+      const metaData = {
+        type: 'website',
+        url: 'https://formester.com/templates/',
+        title: 'The Formester Templates',
+        description:
+          'Use our Formester form templates including surveys, reviews, registrations, & more for any industry! Automate workflows with online templates.',
```
Comment (ankitsinghaniyaz) [inline pages/templates/index.vue:19]: nice touch 👍

File: pages/templates/category/_slug.vue [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        mainImage:
+          'https://formester.com/formester-form-builder-background.png', // need to update with blog page image
+        mainImageAlt: 'Form builder showing drag and drop functionality', // need to update with blog page image alt
+      }
+      return getSiteMeta(metaData)
+    },
+  },
+  head() {
+    return {
+      title: 'Templates | Formester',
```
Comment (ankitsinghaniyaz) [inline pages/templates/category/_slug.vue:29]: but you missed this one?? 

Should instead show the category name here

``` `${category.name} Templates | Formester` ```

File: pages/templates/_slug.vue [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+  mounted() {
+    this.getTemplate()
+  },
+  computed: {
+    meta() {
+      const metaData = {
+        type: 'website',
+        url: `https://formester.com/templates/${this.$route.params.slug}/`,
+        title: this.template?.name || 'Formester',
+        description: this.template?.description,
```
Comment (ankitsinghaniyaz) [inline pages/templates/_slug.vue:53]: should also have a default description which would work with the title incase the description is not coming

**Review by ankitsinghaniyaz — CHANGES_REQUESTED** [review]

File: pages/templates/category/_slug.vue [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+export default {
+  components: { Templates },
+  computed: {
+    meta() {
+      const metaData = {
+        type: 'website',
+        url: 'https://formester.com/templates/',
+        title: 'The Formester Templates',
+        description:
+          'Use our Formester form templates including surveys, reviews, registrations, & more for any industry! Automate workflows with online templates.',
```
[the remark this answers is not in the document]
Comment (anirudh-tantry) [reply pages/templates/category/_slug.vue]: any suggestion for this?

**Review by ankitsinghaniyaz — APPROVED** [review]

**Pull request merged**